File: android_mobile/src/main/java/com/alexstyl/specialdates/analytics/Analytics.java
package com.alexstyl.specialdates.analytics;

import com.alexstyl.specialdates.TimeOfDay;
import com.alexstyl.specialdates.contact.Contact;
import com.alexstyl.specialdates.donate.Donation;
import com.alexstyl.specialdates.events.peopleevents.EventType;

public interface Analytics {
    @Deprecated
    void trackAction(Action action);

    @Deprecated
    void trackAction(ActionWithParameters event);

    void trackScreen(Screen screen);

    void trackAddEventsCancelled();

    void trackEventAddedSuccessfully();

    void trackContactSelected();

    void trackEventDatePicked(EventType eventType);

    void trackEventRemoved(EventType eventType);

    void trackImageCaptured();

    void trackExistingImagePicked();

    void trackAvatarSelected();

    void trackContactUpdated();

    void trackContactCreated();

    void trackDailyReminderEnabled();

    void trackDailyReminderDisabled();

    void trackDailyReminderTimeUpdated(TimeOfDay timeOfDay);

    void trackWidgetAdded(Widget widget);

    void trackWidgetRemoved(Widget widget);

    void trackDonationStarted(Donation donation);

    void trackAppInviteRequested();

    void trackDonationRestored();

    void trackDonationPlaced(Donation donation);

    void trackFacebookLoggedIn();

    void trackOnAvatarBounce();

    void trackFacebookLoggedOut();

    void trackVisitGithub();

    void trackContactDetailsViewed(Contact contact);

    void trackNamedaysScreen();
}

File: memento/src/main/java/com/alexstyl/specialdates/analytics/Analytics.kt
package com.alexstyl.specialdates.analytics

import com.alexstyl.specialdates.TimeOfDay
import com.alexstyl.specialdates.contact.Contact
import com.alexstyl.specialdates.donate.Donation
import com.alexstyl.specialdates.events.peopleevents.EventType

interface Analytics {
    fun trackThemeSelected(string: String)

    fun trackScreen(screen: Screen)

    fun trackAddEventsCancelled()

    fun trackEventAddedSuccessfully()

    fun trackContactSelected()

    fun trackEventDatePicked(eventType: EventType)

    fun trackEventRemoved(eventType: EventType)

    fun trackImageCaptured()

    fun trackExistingImagePicked()

    fun trackAvatarSelected()

    fun trackContactUpdated()

    fun trackContactCreated()

    fun trackDailyReminderEnabled()

    fun trackDailyReminderDisabled()

    fun trackDailyReminderTimeUpdated(timeOfDay: TimeOfDay)

    fun trackWidgetAdded(widget: Widget)

    fun trackWidgetRemoved(widget: Widget)

    fun trackDonationStarted(donation: Donation)

    fun trackAppInviteRequested()

    fun trackDonationRestored()

    fun trackDonationPlaced(donation: Donation)

    fun trackFacebookLoggedIn()

    fun trackOnAvatarBounce()

    fun trackFacebookLoggedOut()

    fun trackVisitGithub()

    fun trackContactDetailsViewed(contact: Contact)

    fun trackNamedaysScreen()

    fun trackDailyReminderTriggered()
}
